[14.0][FIX] stock_available_to_promise_release: unrelease

jbaudoux opened the pull request:
Backport of https://github.com/OCA/wms/pull/807
* Recover initial procure_method
* Before this change, once you have unreleased a move, you can steal whatever is available on output :/

Also ensures that split does never changes the procure_method to 'make to stock'. This was fixed in v15 by odoo.
* This backports https://github.com/odoo/odoo/commit/4180afb95112dbb1119fd68b7bd3f2f5e1160422#diff-55c6314416a6a400da6acd5018d161a55eeeb0e3008fec8828121e3dd12be0ebR1744

cc @TDu @mt-software-de @sebalix

jbaudoux committed:
[FIX] stock_available_to_promise_release: unrelease

Recover initial procure_method

mt-software-de reviewed (approved):
LGTM, can you check the tests?

jbaudoux committed:
[MIG] stock_available_to_promise_release: unrelease

Back-ported from v16.0

jbaudoux commented:
> LGTM, can you check the tests?

Failing due to this fix in odoo not in v14.0 https://github.com/odoo/odoo/commit/4180afb95112dbb1119fd68b7bd3f2f5e1160422#

jbaudoux committed:
[FIX] stock_available_to_promise_release: procure_method

Always keep procure_method on split.

This backports https://github.com/odoo/odoo/commit/4180afb95112dbb1119fd68b7bd3f2f5e1160422#diff-55c6314416a6a400da6acd5018d161a55eeeb0e3008fec8828121e3dd12be0ebR1744

jbaudoux commented:
/ocabot merge patch
